src/components/Comments.jsx

```diff
@@ -1,90 +1,140 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useContext } from 'react';
 import { toast } from 'react-toastify';
-import { getCommentsByArticleId, postCommentsByArticleId } from '../api';
+import {
+	getCommentsByArticleId,
+	postCommentsByArticleId,
+	deleteComment,
+} from '../api';
 import Loading from './Loading';
+import BlogContext from '../contexts/BlogContext';
 
 function Comments({ id }) {
 	const [comments, setComments] = useState([]);
 	const [error, setError] = useState(false);
 	const [isLoading, setIsLoading] = useState(true);
 	const [newComment, setNewComment] = useState({
-    "username": '',
-    "body": ''
-  })
-  const [postConfirmation, setPostConfirmation] = useState(false)
+		username: '',
+		body: '',
+	});
+	const [postConfirmation, setPostConfirmation] = useState(false);
+	const [deleteConfirmation, setDeleteConfirmation] = useState(false);
+	const { user } = useContext(BlogContext);
 
 	useEffect(() => {
+		setError(false);
 		getCommentsByArticleId(id)
 			.then(({ commentList }) => {
 				setComments(commentList);
 				setIsLoading(false);
 			})
 			.catch((err) => {
-				setError(err);
+				setError(true);
 				setIsLoading(false);
 			});
-	}, []);
+	}, [deleteConfirmation]);
```

Comment:
When you delete the comment, you shouldn't need to refetch from the api to account for other users comments - instead you should be deleting it frmo a useState array of comments rendered. kind of like optimistic deletion!